File: Frontend/types/habit.ts
export interface Habit {
  id: string;
  user_id: string;
  log_date: string;
  stress_level: number | null;
  sleep_hours: number | null;
  activity: string | null;
  activity_duration_min: number | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateHabitRequest {
  log_date: string;
  stress_level?: number;
  sleep_hours?: number;
  activity?: string;
  activity_duration_min?: number;
  notes?: string;
}

File: Backend/app/presentation/schemas/habit.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, Field


class HabitCreateRequest(BaseModel):
    log_date: date
    stress_level: int | None = Field(None, ge=1, le=10)
    sleep_hours: float | None = Field(None, ge=0, le=24)
    activity: str | None = Field(None, max_length=100)
    activity_duration_min: int | None = Field(None, ge=1, le=600)
    notes: str | None = None


class HabitUpdateRequest(BaseModel):
    stress_level: int | None = Field(None, ge=1, le=10)
    sleep_hours: float | None = Field(None, ge=0, le=24)
    activity: str | None = Field(None, max_length=100)
    activity_duration_min: int | None = Field(None, ge=1, le=600)
    notes: str | None = None


class HabitResponse(BaseModel):
    id: UUID
    user_id: UUID
    log_date: date
    stress_level: int | None
    sleep_hours: float | None
    activity: str | None
    activity_duration_min: int | None
    notes: str | None
    created_at: datetime
    updated_at: datetime
